heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -311,8 +316,16 @@ public QueryTrace buildTrace() {
                 final List<Callable<AsyncFuture<FetchData.Result>>> fetches = new ArrayList<>();
                 accept(metricBackend -> {
                     for (final Series series : result.getSeries()) {
-                        final Span fetchSeries = tracer.spanBuilderWithExplicitParent(
-                            "localMetricsManager.fetchSeries", fetchSpan).startSpan();
+                        // Without sampling spans this would produce way too many.
+                        final SpanBuilder fetchSeriesSpanBuilder =
+                            tracer.spanBuilderWithExplicitParent(
+                                "localMetricsManager.fetchSeries", fetchSpan);
+                        if (result.getSeries().size() > 100) {
```

Comment:
The series limit should probably also be configurable, but if not it should at least be a class static.

Reply:
You could also move the `result.getSeries().size()` above the for loop to avoid counting the list every iteration.